body_test.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+var allTheBytes []byte = []byte{1, 3, 7, 15, 31, 63, 127, 255}
+
+func TestBodyReading(t *testing.T) {
+
+	filename := "test-files/body-reading"
+
+	_file, _err := os.Create(filename)
```

Comment:
I'm curious why these variables start with an `_`? I'm not familiar with this convention.

Reply:
I didn't really like having them in the function scope, since they're only initializing code.  Maybe I could instead put that block of code into a function and call it... This was just a cleanliness thing

Reply:
Let's chat about this, I'm not really sure what this means. The scope of the variables shouldn't change according to whether they have a `_` prefix or not, and the variables declared inside a function won't ever be visible to code outside of that function.

Reply:
I know that it doesn't change the scope, it was purely a naming thing, I tend to do this because scoping in other lanagues (python, js-var) works a little strangely, so I tend to be careful about giving names that I'm likely to reuse in a nested scope.